File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.py
from collections import deque
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if root is None:
            return []

        result = []
        q = deque([root])

        while len(q) > 0:
            result.append(q[0].val)

            for _ in range(len(q)):
                node = q.popleft()

                if node.right is not None:
                    q.append(node.right)

                if node.left is not None:
                    q.append(node.left)

        return result

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var rightSideView = function (root) {
    const q = [root];
    const result = [];

    while (q.length > 0) {
        const n = q.length;
        if (q[0]) result.push(q[0]?.val);

        for (let i = 0; i < n; i++) {
            const node = q.shift();

            if (node?.right) q.push(node.right);
            if (node?.left) q.push(node.left);
        }
    }

    return result;
};